### PR TITLE
OGM-983. Changes from EPAM

### PR DESCRIPTION
**Pull request by schernolyas**
The PR contains migration to Apache Ignite 2.1

### CODE REVIEW

File: ignite/src/test/resources/skipTests [hunk 1]
```diff
@@ -40,3 +40,9 @@ org.hibernate.ogm.backendtck.inheritance.tableperclass.depositor.PolymorphicAsso
 org.hibernate.ogm.backendtck.inheritance.tableperclass.family.TablePerClassInheritancePersistTest
 org.hibernate.ogm.backendtck.inheritance.singletable.depositor.PolymorphicAssociationTest
 
+//failed tests after migration Ignite 1.9->Ignite 2.1
+org.hibernate.ogm.backendtck.associations.collection.unidirectional.CollectionUnidirectionalTest#testUnidirectionalCollection
+org.hibernate.ogm.backendtck.associations.manytoone.ManyToOneTest#testUnidirectionalManyToOne
+org.hibernate.ogm.backendtck.jpa.JPAResourceLocalTest#testBootstrapAndCRUD
```
Comment (gsmet) [inline ignite/src/test/resources/skipTests:46]: @schernolyas Quick question: maybe we should try to understand why they fail now whereas they were working before?

Comment (schernolyas) [reply ignite/src/test/resources/skipTests:46]: Hi @gsmet  !
About JPAResourceLocalTest ... Reason is problem with em.flush(). Ignite team said it is feature of distributed cache.
About other tests ...Now I don't know. I will investigate.

They worked in Ignite 1.9

File: ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java [hunk 2]
```diff
@@ -189,11 +186,11 @@ private TestCacheConfigBuilder createCacheConfig( String name ) {
 		return new TestCacheConfigBuilder( name );
 	}
 
-	private class TestCacheConfigBuilder {
+	private class TestCacheConfigBuilder implements Serializable {
```
Comment (DavideD) [inline ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java]: Is this necessary?

Comment (schernolyas) [reply ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java]: rechecking

Comment (schernolyas) [reply ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java]: reverted

**Comment by DavideD** [comment]
Can you squash this 3 commits together? Thanks

**Comment by DavideD** [comment]
> Can you squash this 3 commits together? Thanks

I mean, keep the test separated in their own commit and squash the other commits together

File: ignite/src/test/resources/skipTests (hunk 1 above)
Comment (schernolyas) [reply ignite/src/test/resources/skipTests:46]: org.hibernate.ogm.backendtck.associations.manytoone.ManyToOneTest#testUnidirectionalManyToOne **uses session.flush and session.commit**.  it is question for Ignite team

Comment (schernolyas) [reply ignite/src/test/resources/skipTests:46]: With +org.hibernate.ogm.backendtck.associations.collection.unidirectional.CollectionUnidirectionalTest#testUnidirectionalCollection same situation

Comment (DavideD) [reply ignite/src/test/resources/skipTests:46]: > About JPAResourceLocalTest ... Reason is problem with em.flush(). Ignite team said it is feature of distributed cache.

Could you explain a bit more the issue with flush()? It was working before, is something that has changed between the 2 versions?

**Comment by schernolyas** [comment]
Hi @DavideD !

Do you mean squash in the branch?

File: ignite/src/test/resources/skipTests (hunk 1 above)
Comment (schernolyas) [reply ignite/src/test/resources/skipTests:46]: I want know it too.  I will try to take details

**Comment by DavideD** [comment]
> Do you mean squash in the branch?

Yes in this same branch. You can push force here with the commits squashed together.
This way on master we won't have commits reverting other commits and it will look like the mistakes were never there in the first place. This also means that the commit log is easier to read

**Comment by DavideD** [comment]
I mean  branch `schernolyas:OGM-983-EPAM-DEV4`

**Comment by schernolyas** [comment]
mmm git not allow squash three commits in my case.  I will create new branch. It simler.

**Comment by schernolyas** [comment]
changed by https://github.com/hibernate/hibernate-ogm-ignite/pull/6

**Comment by DavideD** [comment]
> mmm git not allow squash three commits in my case. I will create new branch. It simler.

https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Pull request closed without merging**